linalg/solvers.cpp

```diff
@@ -558,13 +558,14 @@ void CGSolver::Mult(const Vector &b, Vector &x) const
       d = r;
    }
    nom0 = nom = Dot(d, r);
+   snom = sqrt(nom);
```

Comment:
I remember discussions about `sqrt` or not for this norm - we should check the reason and if we do want to do this here.

Reply:
`CGSolver` monitoring is definitely wrong. You can check against `PetscPCG` and `HyprePCG`: now the code produces the same unpreconditioned iterates

Reply:
Using the squared norm was intentional, but I agree that we should revisit it and probably change it in all the Krylov solvers.

Lets do it in another PR though. Can you revert so we can approve this one quicker?

Reply:
ping: @stefanozampini

Reply:
@tzanio Dropped the commit, it is here if you want to fix MFEM later https://github.com/mfem/mfem/tree/fix-monitoring

Reply:
Thank you Stefano